xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-ui/src/XWiki/Lightbox/WebHome.xml

```diff
@@ -0,0 +1,838 @@
+        <prettyName>Caching policy</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>long|short|default|forbid</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </cache>
+      <code>
+        <contenttype>PureText</contenttype>
+        <disabled>0</disabled>
+        <editor>PureText</editor>
+        <name>code</name>
+        <number>2</number>
+        <prettyName>Code</prettyName>
+        <rows>20</rows>
+        <size>50</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </code>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <parse>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>parse</name>
+        <number>4</number>
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>long</cache>
+    </property>
+    <property>
+      <code>define('lightboxTranslationKeys', {
+  prefix: 'core.viewers.attachments.',
+  keys: [
+    'date',
+    'author'
+  ]
+});
+
+require(['jquery', 'xwiki-meta', 'xwiki-l10n!lightboxTranslationKeys', 'blueimp-gallery', 'blueimp-gallery-fullscreen', 'blueimp-gallery-indicator'], function($, xm, l10n, gallery) {
+  var myOpenLightbox;
+
+  /*
+   * Make sure that the toolbar will remain open also while hovering it, not just the image.
+   */
+  var keepToolbarOpenOnHover = function(image) {
+    var timeout = setTimeout(function() {
+      image.popover('hide');
+    }, 3000);
+    $('.imageToolbar').on('mouseenter', function() {
+      clearTimeout(timeout);
+    });
+    $('.imageToolbar').parent().on('mouseleave', function() {
+      image.popover('hide');
+    });
```

Comment:
Does this work on mobile, in particular if the image is inside a link?

Reply:
On touch screens the image toolbar is shown when tapping on the image, AFAIK, and hidden when tapping outside. Of course, if the image is inside a link then that link will be activated and the user will leave the current page. This is the behavior we want for the first version at least.

The mouse event listeners will be triggered only on screens were pointer events are supported (desktop). In this case, the image toolbar is shown when hovering the image and hidden when the mouse leaves the image.

Reply:
> The mouse event listeners will be triggered only on screens were pointer events are supported (desktop).

I remember also having this behavior mouse over behavior on websites when touching an image, but maintaining it so that it's not a click. But maybe it's just special touch support and not mouse over emulated by mobile browser, did not dig much.